### DISCUSSION TITLE
Allow changing output device (sinkId) with local playback

### DISCUSSION THREAD

**Discussion by sevmonster** [jcorporation/myMPD#949] [Ideas]
I have not looked into how local playback works, but I believe it uses an `<audio>` element? Would it be possible to change the output device using [`sinkId`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/setSinkId)? It seems this is not supported by all browsers however.

**Comment by jcorporation** [comment]
Good idea.

**Comment by jcorporation** [comment]
I tested it, but I could not get reliable a list of output devices. Without this, the `setSinkId` is unusable.

Prefered approach will be:
```
    navigator.mediaDevices.selectAudioOutput()
        .then( (device) => {
            console.log(device.kind + ": " + device.label + " id = " + device.deviceId);
        })
        .catch(function(error) {
            console.log(error.name + ": " + error.message);
        });
```
But that is not yet support by current browsers: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/selectAudioOutput

Other approach will be:
```
    navigator.mediaDevices.getUserMedia({audio: true, video: false})
        .then(s => {
            navigator.mediaDevices.enumerateDevices().then(devices => {
                devices.forEach(device => {
                    console.log(device.kind + ':' + device.label);
                })
            })
        })
        .catch(error => {
            console.error(error)
        })
```
But I get no output devices in my environment (firefox and chromium on linux).

**Comment by sevmonster** [comment]
That was also my experience. For Chromium, try giving microphone permission to the site, that seems to help.

**Comment by jcorporation** [comment]
This is now implemented in the v22.0.0 branch. It works for me in Firefox on Linux. Can you test it in your setup?
